pkg/cloud/client.go

```diff
@@ -159,7 +159,8 @@ func NewClientFromConf(conf Config, clientConfig *corev1.ConfigMap) (Client, err
 	}
 
 	clientCacheKey := generateClientCacheKey(conf)
-	if client, exists := clientCache.Get(clientCacheKey); exists {
+	if item := clientCache.Get(clientCacheKey); item != nil {
+		var client interface{} = item.Value()
 		return client.(Client), nil
```

Comment:
this is more for my own understanding than a request for change
But, why is casting required if`ttlcache.Cache` is parametrized with the types for key and value?